sdks/python/apache_beam/runners/worker/statesampler.py

```diff
@@ -49,18 +67,43 @@ def __init__(self, prefix, counter_factory,
     self.sampling_period_ms = sampling_period_ms
     super(StateSampler, self).__init__(sampling_period_ms)
 
+    # TODO(pabloem) - Remove this once all clients register the
+    # sampler independently.
+    self.register()
+
   def stop_if_still_running(self):
     if self.started and not self.finished:
       self.stop()
 
+  @staticmethod
+  def simple_tracker():
+    sampler = StateSampler('', CounterFactory())
+    sampler.register()
+    return sampler
+
+  @staticmethod
+  def create_and_register_tracker(
+      prefix, counter_factory, sampling_period_ms=DEFAULT_SAMPLING_PERIOD_MS):
+    sampler = StateSampler(prefix, counter_factory, sampling_period_ms)
+    sampler.register()
+    return sampler
+
+  def register(self):
```

Comment:
Maybe register_on_current_thread? I'd actually have the caller call EXECUTION_STATE_SAMPLERS.set_sampler directly, so the interaction of getting the sampler is symmetrc with setting it.